### ISSUE TITLE
Issues on server to server transfer

### ISSUE THREAD

**Issue by davidbacter01** [HumanBrainProject/pyunicore#32]
Hello!
I was trying to implement a solution for downloading output files straight to an ebrains bucket (https://wiki.ebrains.eu/bin/view/Collabs/the-collaboratory/Documentation%20Bucket/API) and for some reason, any transfer that is created has a status 'FAILED' (regardless of the accessibility of the bucket - private/public) and the file is never uploaded to the bucket.
The way I am trying to achieve this is as follows:
`
 job = unicore_client.Job(transport, job_url)
 job = self.get_job(job_url)
 wd = job.working_dir.listdir()
 transfer = wd[output_file].storage.send_file(output_file, bucket_upload_url)
 print(transfer.properties['status'])
`
Are there any known issues regarding server-to-server transfers or am I doing it wrong?
Thanks!

**Comment by BerndSchuller** [comment]
hi David!

the "server-to-server" here refers to both sides being UNICORE - you cannot use this to send files from UNICORE directly to some generic https server.

What would work instead is doing an upload as a stage-out as part of the job ... if the lifetime of the temporary bucket upload URL is long enough!

So you would add an "Export" to your job.

```
job_description = { ... }
job_exports = []
job_description['Exports'] = job_exports

job_exports.append( {"From": "your_output_file", "To": "https://temp_upload_url" } )
```

the upload will be done after the executable has finished running of course)
 
Another option is to create and use some sort of helper script and do the upload as part of the job itself (e.g. in the post processing stage)

Hope this helps!

Best regards,
Bernd

**Comment by davidbacter01** [comment]
Hi Bernd!

Thank you for your prompt response. This is really helpful!
Would you think the 'UNICORE to some generic HTTPS server transfer' is something to consider for implementation in the foreseeable future by the UNICORE team?
Thanks again for your reply!

Best regards,
David

**Comment by BerndSchuller** [comment]
yes! I think that would be a good feature, should be available in the next regular server release (9.2)
https://github.com/UNICORE-EU/unicorex/issues/6
